lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -234,6 +236,12 @@ def resource_objects_for(serializers)
         [@primary, @included]
       end
 
+      def paginate?
+        !serialization_context.nil? &&
+          serializer.respond_to?(:paginated?) &&
+          serializer.paginated?
```

Comment:
maybe the serializer sometimes doesn't respond to paginated? like.. maybe it's sometimes not an array serializer?